etc/ci-gather-coverage.sh

```diff
@@ -71,7 +71,40 @@ fi;
 # Always output coveralls JSON, as ci-coveralls-merge.py relies
 # on it being present and containing all relevant metadata...
 Print("Outputting JSON for Coveralls...\n");
-OutputCoverallsJsonCoverage(r, "gap-coveralls.json", prefix);
+
+env := GAPInfo.SystemEnvironment;;
+if IsBound(env.GITHUB_ACTIONS) then
```

Comment:
@wilfwilson @ChrisJefferson @ssiccha if you take a look at the hacky patch I made to etc/ci-gather-coverage.sh: that's probably the biggest thing we need, and it should be folded into the profiling package: a way to detect that we are inside a GitHub Action run, and then use that to set the correct meta data for coveralls... But perhaps there is a way to avoid all that, too, e.g. by using https://github.com/marketplace/actions/coveralls-github-action

The main reason these shenigangs are needed is to help coveralls understand which coverage results belong together / belong to a specific PR. But again, perhaps the GitHub Action I just linked to helps us with that and we can simplify the code in profiling. But I am afraid that likely we'll need *some* kind of modification to profiling (also for GAP packages)... Not sure